core/src/ibc/transaction_handler/datagrams.rs

```diff
@@ -93,6 +106,29 @@ impl Encodable for Datagram {
                     .append(client_identifier)
                     .append(counterparty_client_identifier);
             }
+            Datagram::ConnOpenTry {
+                desired_identifier,
+                counterparty_connection_identifier,
+                counterparty_prefix,
+                counterparty_client_identifier,
+                client_identifier,
+                proof_init,
+                proof_consensus,
+                proof_height,
+                consensus_height,
+            } => {
+                s.begin_list(10);
+                s.append(&DatagramTag::ConnOpenTry)
```

Comment:
I think we really need to improve macro to support `enum` and `tuple` someday...